File: backend/src/main/scala/microservices/operations/hotel/objects/RegisterHotelManagerPlannerRequest.scala
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class RegisterHotelManagerPlannerRequest(email: String, displayName: String, hotelName: String, location: String, password: String)
object RegisterHotelManagerPlannerRequest:
  given sourceEncoder: Encoder[RegisterHotelManagerPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[RegisterHotelManagerPlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/hotel/objects/RegisterHotelManagerPlannerRequest.ts
export type RegisterHotelManagerPlannerRequest = {
  email: string
  displayName: string
  hotelName: string
  location: string
  password: string
}

export const registerHotelManagerPlannerRequestFromJson = (json: string): RegisterHotelManagerPlannerRequest =>
  JSON.parse(json) as RegisterHotelManagerPlannerRequest

export const registerHotelManagerPlannerRequestToJson = (value: RegisterHotelManagerPlannerRequest): string =>
  JSON.stringify(value)
